### PR TITLE
UPSTREAM: 14705: Inline fields of SecurityContext into PodSecurityContext

### PR DESCRIPTION
**Pull request by pmorie**
Depends on #5144; patch PR for kubernetes/kubernetes#14705

@pweil- @swagiaal @deads2k @liggitt

### CODE REVIEW

**Comment by pmorie** [comment]
Known issue right now is that the serialization for v1beta3 needs to be worked out re: round-trip testing.

File: pkg/api/serialization_test.go [hunk 1]
```diff
@@ -220,6 +220,14 @@ func fuzzInternalObject(t *testing.T, forVersion string, item runtime.Object, se
 		func(j *deploy.DeploymentConfig, c fuzz.Continue) {
 			c.FuzzNoCustom(j)
 			j.Triggers = []deploy.DeploymentTriggerPolicy{{Type: deploy.DeploymentTriggerOnConfigChange}}
+			fmt.Println(forVersion)
```
Comment (liggitt) [inline pkg/api/serialization_test.go]: t.Log()?

**Comment by liggitt** [comment]
any way to collapse the carry commits?

**Comment by liggitt** [comment]
Does this obsolete https://github.com/openshift/origin/pull/5144?

**Comment by pmorie** [comment]
@liggit -- I would prefer to keep the carries separate for now since there
will be more carries for experimental and v1beta3

It doesn't obsolute 5144 -- builds on it since I anticipated #5144 would go
in first.

File: pkg/api/serialization_test.go (hunk 1 above)
Comment (pmorie) [inline pkg/api/serialization_test.go]: That's coming out.

**Comment by pmorie** [comment]
@liggitt Actually, if we think this is ready to go, I could close #5144 and collapse the carries into single commits per API version, so 1 for experimental, 1 for v1beta3.

**Comment by pmorie** [comment]
Okay, I have closed out #5144 and recut commits.  For future PRs that build on this, we will add revert commits for the carries and new carries that have the right state, so that no squashing or recutting is needed later.

**Comment by pmorie** [comment]
[test]

**Comment by pmorie** [comment]
Closing, going to open a new PR for all volume security changes.

**Pull request closed without merging**